src/hotspot/share/gc/shared/gcConfig.cpp

```diff
@@ -111,6 +111,38 @@ void GCConfig::select_gc_ergonomically() {
   }
 }
 
+void select_gc_ergonomically_dedicated() {
+  julong phys_mem = os::physical_memory();
+  if (os::active_processor_count() <= 1) {
+#if INCLUDE_SERIALGC
+      FLAG_SET_ERGO_IF_DEFAULT(UseSerialGC, true);
+#endif
+    } else if (phys_mem >= 16*G) {
+#if INCLUDE_ZGC
+      FLAG_SET_ERGO_IF_DEFAULT(UseZGC, true);
+#endif
+    } else if (phys_mem > 2048*M) {
+#if INCLUDE_G1GC
+      FLAG_SET_ERGO_IF_DEFAULT(UseG1GC, true);
+#endif
+    } else { 
+#if INCLUDE_PARALLELGC
+      FLAG_SET_ERGO_IF_DEFAULT(UseParallelGC, true);
+#endif
+    }
+}
+
+void GCConfig::select_gc_ergonomically() {
+  if (strcmp(ErgonomicsProfile, "shared") == 0) {
+    select_gc_ergonomically_shared();
+  } else if (strcmp(ErgonomicsProfile, "dedicated") == 0) {
+    select_gc_ergonomically_dedicated();
+  } else {
+    // We have got to have at least one GC selected. Otherwise, we will crash.
+    ShouldNotReachHere();
+  }
```

Comment:
Not sure I understand your comment on single cores. The algorithm is still the same if `shared`, and if it is `dedicated`, it will still be Serial GC if it is a single active processor no matter how much memory.